### PR TITLE
boards: arm: add support for NUCLEO-L433RC-P

### PR DESCRIPTION
**Pull request by mtudan**
Support the ST STM32 Nucleo-64 development board with
STM32L433RC SoC.
Tested samples: hello_world, blinky and button.

Signed-off-by: Matija Tudan <mtudan@mobilisis.hr>

### CODE REVIEW

File: boards/arm/nucleo_l433rc_p/nucleo_l433rc_p_defconfig [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+CONFIG_CLOCK_STM32_AHB_PRESCALER=1
+CONFIG_CLOCK_STM32_APB1_PRESCALER=1
+CONFIG_CLOCK_STM32_APB2_PRESCALER=1
+
+# console
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+#enable pwm
+CONFIG_PWM=y
```
Comment (erwango) [inline boards/arm/nucleo_l433rc_p/nucleo_l433rc_p_defconfig:44]: Peripheral enabling is up to application responsibility. Please remove

File: boards/arm/nucleo_l433rc_p/nucleo_l433rc_p.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  - gnuarmemb
+  - xtools
+ram: 64
+flash: 256
+supported:
+  - nvs
+  - pwm
+  - can
+  - counter
+  - spi
```
Comment (erwango) [inline boards/arm/nucleo_l433rc_p/nucleo_l433rc_p.yaml:16]: - i2c
- arduino_i2c
- arduino_spi

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for this addition!
This looks globally fine, aside from few requested changes.

**Review by ioannisg — APPROVED** [review]
Looks good to me, apart from the issues mentioned by @erwango

File: boards/arm/nucleo_l433rc_p/nucleo_l433rc_p_defconfig (hunk 1 above)
Comment (mtudan) [reply boards/arm/nucleo_l433rc_p/nucleo_l433rc_p_defconfig]: Fixed. Just FYI, I used nucleo_l432kc board as a reference for this file. So maybe remove it there as well...

File: boards/arm/nucleo_l433rc_p/nucleo_l433rc_p.yaml (hunk 2 above)
Comment (mtudan) [reply boards/arm/nucleo_l433rc_p/nucleo_l433rc_p.yaml:16]: Added.

File: boards/arm/nucleo_l433rc_p/nucleo_l433rc_p_defconfig (hunk 1 above)
Comment (erwango) [reply boards/arm/nucleo_l433rc_p/nucleo_l433rc_p_defconfig]: ok, txs. I'll fix that.

File: boards/arm/nucleo_l433rc_p/nucleo_l433rc_p.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+identifier: nucleo_l433rc
```
Comment (erwango) [inline boards/arm/nucleo_l433rc_p/nucleo_l433rc_p.yaml:1]:
```suggestion
identifier: nucleo_l433rc_p
```

**Review by erwango — CHANGES_REQUESTED** [review]
Late finding.
Last change to be made and we're good to go.

File: boards/arm/nucleo_l433rc_p/nucleo_l433rc_p.dts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+		sw0 = &user_button;
+	};
+};
+
+&usart1 {
+	pinctrl-0 = <&usart1_tx_pa9 &usart1_rx_pa10>;
+	current-speed = <115200>;
+};
+
+&usart2 {
```
Comment (FRASTM) [inline boards/arm/nucleo_l433rc_p/nucleo_l433rc_p.dts:51]: on nucleo board MB1319, the **LPUART1** pins PA2, PA3 are connected to the STLink console

**Comment by FRASTM** [comment]
is this PR completing  #26309 ?

File: boards/arm/nucleo_l433rc_p/nucleo_l433rc_p.yaml (hunk 3 above)
Comment (mtudan) [reply boards/arm/nucleo_l433rc_p/nucleo_l433rc_p.yaml]: Good catch, fixed.

**Comment by mtudan** [comment]
> is this PR completing #26309 ?
I didn't notice that PR before... But yes, it seems those are the same boards.

File: boards/arm/nucleo_l433rc_p/nucleo_l433rc_p.dts (hunk 4 above)
Comment (mtudan) [reply boards/arm/nucleo_l433rc_p/nucleo_l433rc_p.dts]: Fixed.

**Review by erwango — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**